#!/usr/bin/env python
PACKAGE = 'spinnaker_driver_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('exposure_time', double_t, 0, "The exposure time for the cameras", 5000.0, 10.0, 15000.0)
gen.add('gain', double_t, 0, "The gain for the cameras", 0.0, 0.0, 15.0)
gen.add('fps', double_t, 0, "The frame rate for the cameras", 45.0, 1.0, 5.0)
gen.add('resize_factor', double_t, 0, "Published image resize factor", 4.0, 1.0, 8.0)
gen.add('resize_images', bool_t, 0, "Resize published images", False)

exit(gen.generate(PACKAGE, 'camera_manager', 'stereoCameraParameters'))